tests/req_tests.rs

```diff
@@ -57,3 +57,31 @@ fn response_neq_test() {
 
     assert_eq!(true, original != request)
 }
+
+// Tests for parsing the query string to params and args
+#[test]
+fn test_params_from_route() {
+    let mut request = Request::new().with_route("/test/route/query?arg1=1&arg2=2".to_string());
+    let query: Vec<&str> = ["query", "arg1=1&arg2=2"].to_vec();
+    request = request.params_from_route(query);
+    assert_eq!(request.get_params(), ["param".to_string()].to_vec())
+}
+
+#[test]
+fn test_multiple_params_from_route() {
+    let mut request = Request::new()
+                              .with_route("/test/route/param1?param2?arg1=1&arg2=2".to_string());
+    let query: Vec<&str> = ["param1", "param2", "arg1=1&arg2=2"].to_vec();
+    request = request.params_from_route(query);
+    assert_eq!(request.get_params() , ["param1".to_string(), "param2".to_string()].to_vec())
+}
+
+#[test]
+fn test_empty_params_from_route() {
+    let mut request = Request::new()
```

Comment:
have we tested for passing in no parameters? It seems like the fn might break in that case